pos_order_mgmt/static/src/js/widgets.js

```diff
@@ -0,0 +1,404 @@
+                temporary: !order_data.return,
+            });
+            // Set Generic Info
+            if (!order_data.return) {
+                order.name = order_data.pos_reference || order_data.name;
+            }
+            if (order_data.partner_id.length) {
+                order_data.partner_id = order_data.partner_id[0];
+            }
+            order.set_client(this.pos.db.get_partner_by_id(order_data.partner_id));
+            // Set fiscal position
+            if (order_data.fiscal_position && this.pos.fiscal_positions) {
+                var fiscal_positions = this.pos.fiscal_positions;
+                order.fiscal_position = fiscal_positions.filter(function (p) {
+                    return p.id === order_data.fiscal_position;
+                })[0];
+                order.trigger('change');
+            }
+            // Set order lines
+            var orderLines = order_data.line_ids || order_data.lines || [];
+            self._prepare_orderlines_from_order_data(order, order_data, orderLines);
+            if (order_data.return) {
+                order.return = true;
+                // A credit note should be emited if there was an invoice
+                order.set_to_invoice(order_data.origin_invoice_id);
+                // We'll refunded orders once they are synced
+                order.returned_order_id = order_data.id || order_data.name;
+                order.origin_name = order_data.pos_reference || order.returned_order_id;
+                return order;
+            }
+            if (order_data.returned_order_id) {
+                order.origin_name = order_data.returned_order_id;
+            }
+            order.formatted_validation_date = moment(order_data.date_order).format('YYYY-MM-DD HH:mm:ss');
+            // Set Payment lines
+            var paymentLines = order_data.statement_ids || [];
+            _.each(paymentLines, function (paymentLine) {
+                var line = paymentLine;
+                // In case of local data
+                if (line.length === 3) {
+                    line = line[2];
+                }
+                _.each(self.pos.cashregisters, function (cashregister) {
+                    if (cashregister.id === line.statement_id) {
+                        if (line.amount > 0) {
+                            // If it is not change
+                            order.add_paymentline(cashregister);
+                            order.selected_paymentline.set_amount(line.amount);
+                        }
+                    }
+                });
+            });
+            return order;
+        },
+        _prepare_orderlines_from_order_data: function(order, order_data, orderLines) {
+            _.each(orderLines, function(line) {
+                // In case of local data
+                if (line.length === 3) {
+                    line = line[2];
+                }
+                var product = self.pos.db.get_product_by_id(line.product_id);
```

Comment:
Is missing at the beginning var self = this;
So this line throws the error: `Cannot read property 'db' of undefined`